### PR TITLE
[Fix] Allow to define a custom release name for source-maps upload

### PR DESCRIPTION
**Pull request by retyui**
Fix an issue: https://github.com/getsentry/sentry-react-native/issues/1229

Android cli PR: https://github.com/getsentry/sentry-react-native/pull/1548

### CODE REVIEW

File: src/commands/react_native_xcode.rs [hunk 1]
```diff
@@ -274,15 +274,19 @@ pub fn execute(matches: &ArgMatches<'_>) -> Result<(), Error> {
...
 
-        let release = api.new_release(
+        let default_release_name = format!(
+            "{}@{}+{}",
+            plist.bundle_id(),
+            plist.version(),
+            plist.build()
+        );
+
+        let release_name = env::var("SENTRY_RELEASE_NAME").unwrap_or(default_release_name);
```
Comment (jennmueng) [inline src/commands/react_native_xcode.rs:284]:
```suggestion
        let release_name = env::var("SENTRY_RELEASE").unwrap_or(default_release_name);
```
Let's call it `SENTRY_RELEASE`.

**Review by jennmueng — COMMENTED** [review]
Again, thanks for this! Let's call it `SENTRY_RELEASE` and `SENTRY_DIST`. We would also need the dist to be configurable via setting the `SENTRY_DIST` environment variable too and this should be good to go!

cc @bruno-garcia

File: src/commands/react_native_xcode.rs (hunk 1 above)
Comment (retyui) [reply src/commands/react_native_xcode.rs:284]: done

File: src/commands/react_native_xcode.rs [hunk 2]
```diff
@@ -274,15 +274,20 @@ pub fn execute(matches: &ArgMatches<'_>) -> Result<(), Error> {
...
+        let default_release_name = format!(
+            "{}@{}+{}",
+            plist.bundle_id(),
+            plist.version(),
+            plist.build()
+        );
+
+        let release_name = env::var("SENTRY_RELEASE").unwrap_or(default_release_name);
+        
+
```
Comment (jennmueng) [inline src/commands/react_native_xcode.rs:286]:
```suggestion
```
Test failed due to extra line break

**Comment by retyui** [comment]
Test for macOS failed (but Linux and Windows test passed successfully) is it ok?

**Review by bruno-garcia — COMMENTED** [review]
Do we have any tests that validate this behavior?
@kamilogorek might have thoughts here though

**Comment by jennmueng** [comment]
@retyui The Xcode script would only be run on macOS, this means it does not work. Can you verify that it works locally?

**Comment by retyui** [comment]
>  Can you verify that it works locally?


I will try)

1) Build `cargo build --release --locked --target=x86_64-apple-darwin`
2) Copy `target/x86_64-apple-darwin/release/sentry-cli` to `node_modules/@sentry/cli/sentry-cli`
3) Store in repo `git add -f node_modules/@sentry/cli/sentry-cli ; git commit`
4) Update package.json (to use binary from repo) `"postinstall": "git checkout -f -- :/node_modules"`
5) Push to trigger build on CI


^^^^ Could you please verify that I build binary it correctly? @jennmueng

**Comment by retyui** [comment]
Seems it works as expected

```sh
export SENTRY_RELEASE=1.22.333
```

Снимок экрана 2021-06-02 в 23 11 17

**Comment by retyui** [comment]
I added a small fix to handle `SENTRY_DIST` env


Now it works fine

```
export SENTRY_DIST=777
```

image

**Comment by retyui** [comment]
@bruno-garcia @jennmueng cc

**Comment by kamilogorek** [comment]
Instead of using env vars, this PR should add a flag called `release` which can be used instead. This way, we'll make the experience inline with the one we already have for Gradle. Also overriding `dist` is not necessary, as we already do have `--dist` flag that you can use.

**Comment by retyui** [comment]
> Also overriding dist is not necessary, as we already do have --dist flag that you can use.

You have to understand that CLI script will be run by the build phase "Bundle React Native code and images". 
 `../node_modules/@sentry/cli/bin/sentry-cli react-native xcode` in the xCode

And to define `--dist` need to path that script or use env that I propose to do

File: src/commands/react_native_xcode.rs [hunk 3]
```diff
@@ -274,15 +274,17 @@ pub fn execute(matches: &ArgMatches<'_>) -> Result<(), Error> {
         processor.rewrite(&[base.parent().unwrap().to_str().unwrap()])?;
         processor.add_sourcemap_references()?;
 
+        let default_dist = env::var("SENTRY_DIST").unwrap_or(plist.build().to_string());
+
+        let default_release_name =
+            format!("{}@{}+{}", plist.bundle_id(), plist.version(), default_dist);
+
+        let release_name = env::var("SENTRY_RELEASE").unwrap_or(default_release_name);
```
Comment (kamilogorek) [inline src/commands/react_native_xcode.rs:282]:
```suggestion
        let dist = env::var("SENTRY_DIST").unwrap_or(plist.build().to_string());
        let release_name = env::var("SENTRY_RELEASE").unwrap_or(format!("{}@{}+{}", plist.bundle_id(), plist.version(), dist));
```

File: src/commands/react_native_xcode.rs [hunk 4]
```diff
@@ -294,7 +296,7 @@ pub fn execute(matches: &ArgMatches<'_>) -> Result<(), Error> {
                     org: &org,
                     project: Some(&project),
                     release: &release.version,
-                    dist: Some(&plist.build()),
+                    dist: Some(&default_dist),
```
Comment (kamilogorek) [inline src/commands/react_native_xcode.rs:299]:
```suggestion
                    dist: Some(&dist),
```

File: src/commands/react_native_xcode.rs [hunk 5]
```diff
@@ -304,7 +306,7 @@ pub fn execute(matches: &ArgMatches<'_>) -> Result<(), Error> {
                         org: &org,
                         project: Some(&project),
                         release: &release.version,
-                        dist: Some(dist),
+                        dist: Some(&env::var("SENTRY_DIST").unwrap_or(dist.to_string())),
```
Comment (kamilogorek) [inline src/commands/react_native_xcode.rs:309]:
```suggestion
                        dist: Some(dist),
```

**Review by kamilogorek — CHANGES_REQUESTED** [review]
In this case, flags should always have precedence over env variables. Because of that, `SENTRY_DIST` should be applied only if `--dist` is absent.

File: src/commands/react_native_xcode.rs (hunk 3 above)
Comment (kamilogorek) [reply src/commands/react_native_xcode.rs:282]: This will probably fail formatting :3

**Review by kamilogorek — APPROVED** [review]

**Comment by kamilogorek** [comment]
That's what I thought. Thanks `rustfmt` :S Looks good, will merge once formatted is happy.

**Comment by retyui** [comment]
@kamilogorek seems it needs to start CI manually

**Pull request merged**